tests/eth2/core/beacon/test_committee_helpers.py

```diff
@@ -1,158 +1,61 @@
-    active_validator_count,
+def test_get_committee_count_at_slot(
+    validator_count,
     slots_per_epoch,
     target_committee_size,
-    shard_count,
+    max_committees_per_slot,
     expected_committee_count,
+    genesis_state,
 ):
-    assert expected_committee_count == get_committee_count(
-        active_validator_count=active_validator_count,
-        shard_count=shard_count,
+    state = genesis_state
+    assert expected_committee_count == get_committee_count_at_slot(
+        state,
+        state.slot,
+        max_committees_per_slot=max_committees_per_slot,
         slots_per_epoch=slots_per_epoch,
         target_committee_size=target_committee_size,
     )
 
 
-@pytest.mark.parametrize(
-    (
-        "validator_count,"
-        "slots_per_epoch,"
-        "target_committee_size,"
-        "shard_count,"
-        "expected_shard_delta,"
-    ),
-    [
-        # SHARD_COUNT - SHARD_COUNT // SLOTS_PER_EPOCH
-        (1000, 25, 5, 50, 50 - 50 // 20),
-        # active_validator_count // SLOTS_PER_EPOCH // TARGET_COMMITTEE_SIZE
-        (500, 20, 10, 100, 40),
-    ],
-)
-def test_get_shard_delta(genesis_state, expected_shard_delta, config):
-    state = genesis_state
-    epoch = state.current_epoch(config.SLOTS_PER_EPOCH)
-
-    assert get_shard_delta(state, epoch, config) == expected_shard_delta
-
-
-@pytest.mark.parametrize(
-    (
-        "validator_count,"
-        "slots_per_epoch,"
-        "target_committee_size,"
-        "shard_count,"
-        "current_epoch,"
-        "target_epoch,"
-        "expected_epoch_start_shard,"
-    ),
-    [
-        (1000, 25, 5, 50, 3, 2, 2),
-        (1000, 25, 5, 50, 3, 3, 0),
-        (1000, 25, 5, 50, 3, 4, 48),
-        (1000, 25, 5, 50, 3, 5, None),
-    ],
-)
-def test_get_start_shard(
-    genesis_state, current_epoch, target_epoch, expected_epoch_start_shard, config
-):
-    state = genesis_state.copy(
-        slot=compute_start_slot_of_epoch(current_epoch, config.SLOTS_PER_EPOCH)
-    )
-
-    if expected_epoch_start_shard is None:
-        with pytest.raises(ValidationError):
-            get_start_shard(state, target_epoch, CommitteeConfig(config))
-    else:
-        epoch_start_shard = get_start_shard(
-            state, target_epoch, CommitteeConfig(config)
-        )
-        assert epoch_start_shard == expected_epoch_start_shard
-
-
 SOME_SEED = b"\x33" * 32
 
 
-def test_find_proposer_in_committee(genesis_validators, config):
-    epoch = random.randrange(config.GENESIS_EPOCH, 2 ** 64)
+def test_compute_proposer_index(genesis_validators, config):
     proposer_index = random.randrange(0, len(genesis_validators))
 
     validators = tuple()
-    # NOTE: validators supplied to ``_find_proposer_in_committee``
+    # NOTE: validators supplied to ``compute_proposer_index``
     # should at a minimum have 17 ETH as ``effective_balance``.
     # Using 1 ETH should maintain the same spirit of the test and
     # ensure we can know the likely candidate ahead of time.
```

Comment:
nitpick but this now really is:

```suggestion
     # ensure we can know the candidate ahead of time.
```